### PR TITLE
Add diabetes registration and follow up graph

### PR DESCRIPTION
**Pull request by govindkrjoshi**
**Story card:** [sc-7780](https://app.shortcut.com/simpledotorg/story/7780/add-registrations-and-follow-ups-graph)

## Because

We want to show cumulative and monthly diabetes registrations and follow ups in the dashboard

## This addresses

- Region Summary and Region Summary Schema report cumulative registrations, monthly registrations and monthly follow ups
- Repository and Repository presenter present data to the controller
- The diabetes view adds html to display the Registrations and follow up diabetes graph

## Test instructions

- Turn on diabetes_management_reports on flipper
- Visit the reports page of a facility which has enable_diabetes_management set to true
- Click on Diabetes link
- You should see the Registrations and follow-ups graph

image

### CODE REVIEW

**Review by prabhanshuguptagit — APPROVED** [review]
:shipit: when comments are addressed.

File: db/views/reporting_facility_states_v08.sql [hunk 1]
```diff
@@ -122,10 +122,10 @@ WITH
...
 
     monthly_diabetes_follow_ups AS (
-    SELECT facility_id, month_date, COUNT(distinct(patient_id)) as follow_ups
-    FROM reporting_patient_follow_ups
-    WHERE diabetes = 'yes'
-    GROUP BY facility_id, month_date
+        SELECT facility_id, month_date, COUNT(distinct(patient_id)) as follow_ups
+        FROM reporting_patient_follow_ups
+        WHERE diabetes = 'yes'
+        GROUP BY facility_id, month_date
```
[the remark this answers is not in the document]
Comment (govindkrjoshi) [reply db/views/reporting_facility_states_v08.sql:128]: Added the whitespace on purpose, to be consistent with rest of the file.

File: app/assets/javascripts/common/reports.js [hunk 2]
```diff
@@ -890,6 +892,213 @@ Reports = function (withLtfu) {
     }
   };
 
+  this.setupCumulativeDiabetesRegistrationsGraph = (data) => {
```
[the remark this answers is not in the document]
Comment (PriyangaPKini) [reply app/assets/javascripts/common/reports.js:895]: fixed

File: app/assets/javascripts/common/reports.js [hunk 3]
```diff
@@ -890,6 +892,213 @@ Reports = function (withLtfu) {
...
+            fontColor: this.darkGreyColor,
+            fontSize: 10,
+            fontFamily: "Roboto",
+            padding: 8,
+            min: 0,
+            beginAtZero: true,
+            stepSize: monthlyDiabetesFollowupsYAxis.stepSize,
+            max: monthlyDiabetesFollowupsYAxis.max,
+            callback: (label) => {
+              return this.formatNumberWithCommas(label);
```
[the remark this answers is not in the document]
Comment (govindkrjoshi) [reply app/assets/javascripts/common/reports.js:1045]: There are some doubts we want to clarify with the design team about this. We'll track this as a separate task in the clean up story.

**Pull request merged**